### ISSUE TITLE
Installer version not Working

### ISSUE THREAD

**Issue by Peachey123** [rocksdanister/lively#458]
So I have downloaded the program from the windows store and that version works fine. I wanted to try running a unity program in the background but it said it was not available to do that in that version so I downloaded this version but it won't launch whatsoever. Please help

**Comment by rocksdanister** [comment]
Fill the issue template:
**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Screenshots/Video**
If applicable, add screenshots/video to help explain your problem.

**Desktop (please complete the following information):**
 - OS: [e.g. Windows 10 1909]

**Additional context**
Add any other context about the problem here.

**Log file (Important)**
Share the diagnostic files by:
Open Lively settings (Gear icon) -> Misc. -> Log File
or 
Navigate to Appdata folder `C:\Users\<UserName>\AppData\Local\Lively Wallpaper\logs`

**Comment by TerminatedProcess** [comment]
same for me.. I put this in the reddit group but just removed it as it belongs here..

I uninstalled Lively from my windows 10 (all updates done), deleted the local/programs/lively folder, and downloaded this from your web page ((https://github.com/rocksdanister/lively/releases) . Not the windows store version but the top link. It does not run. No messages, no errors, no task bar icon. I uninstalled it again and downloaded the microsoft store version and it runs fine. In case this is relevant..

b:\>dotnet --info
.NET SDK (reflecting any global.json):
Version: 5.0.202
Commit: db7cc87d51
Runtime Environment:
OS Name: Windows
OS Version: 10.0.19042
OS Platform: Windows
RID: win10-x64
Base Path: C:\Program Files (x86)\dotnet\sdk\5.0.202\
Host (useful for support):
Version: 5.0.5
Commit: 2f740adc14

.NET SDKs installed:
5.0.202 [C:\Program Files\dotnet\sdk]
5.0.202 [C:\Program Files (x86)\dotnet\sdk]

.NET runtimes installed:
Microsoft.AspNetCore.All 2.1.27 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
Microsoft.AspNetCore.App 2.1.27 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.1.14 [C:\Program Files (x86)\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.1.14 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.5 [C:\Program Files (x86)\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 2.1.27 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 3.1.14 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.5 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 3.1.14 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 5.0.5 [C:\Program Files (x86)\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 5.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Also this is my path.. (just to show you its on the path)

C:\Program Files\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\VMware\VMware Player\bin\;C:\Program Files (x86)\Razer Chroma SDK\bin;C:\Program Files\Razer Chroma SDK\bin;C:\Program Files (x86)\Razer\ChromaBroadcast\bin;C:\Program Files\Razer\ChromaBroadcast\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\Mullvad VPN\resources;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\WINDOWS\system32\config\systemprofile\AppData\Local\Microsoft\WindowsApps;C:\Program Files\Java\jdk-16\bin;C:\Users\mryan\kotlinc\bin;C:\Program Files (x86)\dotnet\;%appdata%\Local\Android\Sdk\platform-tools\adb.exe;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files (x86)\Intel\Intel® Memory and Storage Tool\;C:\Users\mryan\AppData\Local\Microsoft\WindowsApps;;C:\Users\mryan\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\mryan\.dotnet\tools

I'm providing this info because it works with ms store, but not the installer so I thought maybe it has to do with the dot net version as MS store version might have a docker like compartmentalization of all needed resources? Just a guess really..

**Comment by rocksdanister** [comment]
The included .net core 3.1 runtime with the setup is all that's needed..
I just tested a fresh install on Windows 10 sandbox and its working.. can you try it there?

I think this is a xaml island related issue (one of the main difference between store and installer version.)

**Comment by TerminatedProcess** [comment]
I downloaded from the github link I posted above (https://github.com/rocksdanister/lively/releases) and it still doesn't work. I uninstalled first. I notice the installer from the prior download and the latest have the same byte size and version number (in the file name). Is this the correct link to test with? I didn't see any other links to try..

**Comment by rocksdanister** [comment]
Yes, there is only one installer version.. I was referring to [Windows 10 Sandbox.](https://www.howtogeek.com/399290/how-to-use-windows-10s-new-sandbox-to-safely-test-apps/)

So not even logfiles are created?
`C:\Users\<UserName>\AppData\Local\Lively Wallpaper\logs`

**Comment by TerminatedProcess** [comment]
Ok I did the following.. no success so far
1. Checked for the logfile directory. Not found.
2. Opened task manager and ran Lively.. but didn't see an entry or even a brief entry.
3. Uninstalled. Opened regedit and went through and deleted all entries with lively or rocksdanister in it. 
4. Reinstalled and still not running. 

Do you know if other users are having this issue or just me? any other ideas to try? At one point I did download your git code and try to compile (not really a C# guy). I didn't get it to go anywhere and I had other priorities. It's a very nice project and I thought I would learn from it. That's why I went through regedit. I did find a huge number of keys left behind after uninstalling. Is that normal?

**Comment by rocksdanister** [comment]
I am not really making any key apart from startup entry and other stuff Inno setup script will do by default, all the settings are stored in file.

Some has issue.. its pretty rare, I just wanted to know if it worked in windows sandbox environment since its clean.

If you run it in visual studio under debug mode it should show error unless uwp needs some special thing I don't know of.

**Comment by TerminatedProcess** [comment]
ok gotcha, I have professional so I'll try sandbox and let  you know..

**Comment by TerminatedProcess** [comment]
Ok so it worked fine under WIndows Sandbox. Cool feature, havent run into it before. Since this is a rare issue, it works in sandbox, and I can use the windows store version.. perhaps it's not worth investing any more effort into this. But I'm available to try stuff if you have anything you want to try. Thanks!

**Comment by rocksdanister** [comment]
>Since this is a rare issue, it works in sandbox, and I can use the windows store version

Its not that simple though, the store version is a completely [separate project](https://github.com/rocksdanister/lively/tree/ms-store) and I implement all the changes from the [main branch](https://github.com/rocksdanister/lively/tree/dev-v1.0-fluent-netcore/src) to it manually.. its very tiring and I might just ditch and switch to the main program in the future if when I break 😭

> But I'm available to try stuff if you have anything you want to try. Thanks!

Running it in visual studio in debug mode should show the error? Once you have the correct tools installed it should compile without any issue:
https://github.com/rocksdanister/lively/wiki/Building

Another thing you can do is have a look at the crash memory dumps, it should be under `..AppData\Local\CrashDumps` - there should be a `livelywpf.exe` file in there.
I never worked with memory dumps before, since I didn't create that file I don't know what's in it.. probably best to email it instead of publishing it here (if interested.)

If the cause can be identified and its not from my side then I can directly report to [xaml island repo.](https://github.com/microsoft/microsoft-ui-xaml)

**Comment by wang526125649** [comment]
So what should I do?
I've tried a lot of ways that I can't open

**Comment by rocksdanister** [comment]
@wang526125649  Try the windows store version of Lively?

**Comment by wang526125649** [comment]
Windows store version, I use the compact version, can't download

> @ wang526125649 Try the windows store version of Lively?

I use the compact version, can't download

**Comment by rocksdanister** [comment]
@wang526125649 
I don't really understand if this is some chinese region issue..
You can try downloading the store msix file [directly.](https://store.rg-adguard.net/)
If that does not work, there is not much I can do unless this issue is resolved:


> > Since this is a rare issue, it works in sandbox, and I can use the windows store version
> 
> Its not that simple though, the store version is a completely [separate project](https://github.com/rocksdanister/lively/tree/ms-store) and I implement all the changes from the [main branch](https://github.com/rocksdanister/lively/tree/dev-v1.0-fluent-netcore/src) to it manually.. its very tiring and I might just ditch and switch to the main program in the future if when I break 😭
> 
> > But I'm available to try stuff if you have anything you want to try. Thanks!
> 
> Running it in visual studio in debug mode should show the error? Once you have the correct tools installed it should compile without any issue:
> https://github.com/rocksdanister/lively/wiki/Building
> 
> Another thing you can do is have a look at the crash memory dumps, it should be under `..AppData\Local\CrashDumps` - there should be a `livelywpf.exe` file in there.
> I never worked with memory dumps before, since I didn't create that file I don't know what's in it.. probably best to email it instead of publishing it here (if interested.)
> 
> If the cause can be identified and its not from my side then I can directly report to [xaml island repo.](https://github.com/microsoft/microsoft-ui-xaml)

If you have more questions please create a new issue as this is off topic to the current discussion.

**Comment by TerminatedProcess** [comment]
I'll look into it.. I couldn't build it before but with the instructions I'll give it a shot. NO crash dumps were produced during installation and running. I also quit Malwarebytes and tried running but no success. Just a guess really. I'll look into compiling now..

**Comment by TerminatedProcess** [comment]
So I got my computer setup per  your build instructions and when I tried running it, I get the following..

![image](https://user-images.githubusercontent.com/63432380/116642695-77075300-a93d-11eb-8efa-7e58674bff3f.png)

I tried continue a few times and ended up here..

![image](https://user-images.githubusercontent.com/63432380/116642733-87b7c900-a93d-11eb-9111-0974739a460a.png)

Next I tried doing this from the project manager, right-clicking on the livelywfp project and stepping into a debug session. I got the following..

![image](https://user-images.githubusercontent.com/63432380/116642998-0f9dd300-a93e-11eb-9ca4-129044451a45.png)

I'm not a c# developer, just been exposed to it here and there over the years and can figure things out. But I'm unsure if I just didn't setup my dev environment correctly (I downloaded things using your build instructions and did a git close) or we are encountering the actual error itself.  Advice?

**Comment by rocksdanister** [comment]
> we are encountering the actual error itself.

I don't think so.. I think this is dev env issue.. are you using the latest visual studio 2019? did all the nuget packages got restored correctly.. is visual studio setup to use package references for nuget libraries..

Also try adding an exception event for the uwp code like shown below and see if anything shows up:
https://github.com/rocksdanister/lively/blob/26db98ed856d3d95265eee99b2ca94c735eb9fce/src/livelywpf/livelywpf/Program.cs#L89-L97

```csharp
           using (var uwp = new rootuwp.App())
           {
                    uwp.UnhandledException += Uwp_UnhandledException;
                    //uwp.RequestedTheme = Windows.UI.Xaml.ApplicationTheme.Light;
                    livelywpf.App app = new livelywpf.App();
                    app.InitializeComponent();
                    app.Startup += App_Startup;
                    app.SessionEnding += App_SessionEnding;
                    app.Run();
           }
           .......
           private static void Uwp_UnhandledException(object sender, Windows.UI.Xaml.UnhandledExceptionEventArgs e)
           {
                   System.Diagnostics.Debug.WriteLine(e.ToString());
           }
```

**Comment by TerminatedProcess** [comment]
Yes to latest 2019 VS.. I'll look at the nugets packages but there were no warnings.. I'm not sure what package references for nuget libraries are. I can google this tomorrow. Also, after installing all of these build instruction libraries, and doing an windows update, my hyper-v was disabled. I noticed because my ubuntu VMWare was suddenly failing.

![image](https://user-images.githubusercontent.com/63432380/116646689-b6866d00-a946-11eb-9502-96bd6c2856f0.png)

Once I turned it back on, your project gives me different errors. Just mentioning in case this is something obvious to you. Google tells me I have a framework issue..

![image](https://user-images.githubusercontent.com/63432380/116646767-e6357500-a946-11eb-9296-a5961d350dd5.png)

![image](https://user-images.githubusercontent.com/63432380/116646790-f0f00a00-a946-11eb-9194-626c2b3d6610.png)

I'll have to pick this up tomorrow.. :)

**Comment by rocksdanister** [comment]
>. Also, after installing all of these build instruction libraries, and doing an windows update, my hyper-v was disabled. I noticed because my ubuntu VMWare was suddenly failing.

[Windows sandbox conflict](https://superuser.com/questions/1442766/virtualbox-fails-to-start-with-verr-nem-vm-create-failed)

Install it through visual studio installer
![Screenshot 2021-04-30 105744](https://user-images.githubusercontent.com/17554161/116652500-ed8a6c00-a9a2-11eb-9c7c-3697bb9b3659.png)

**Comment by TerminatedProcess** [comment]
Ok some progress. I'll give you details just in case something is relevant

1. I uninstalled all of that independent stuff I installed recently (rollback). Uninstalled VS 2019 community.
2. Installed VS 2019 Pro (30 days license)
3. Setup per your screen shot above.
4. Got a clean copy of project via git.
5. VS asked me to install Net Core 3.1.  Done. (x64) (x86 worked as well)
6. Did a build and run and got the setup screen (working). Quit.
7. Re-installed installer from web site. Didn't work.
8. Deployed project and from deploy folder, just copied the exe (not the dll) and dropped it into the lively install folder. Now the installed version works.

Think I'll stop here. Let me know if you want me to further try something.. Thanks!
